src/main.rs

```diff
@@ -43,6 +46,18 @@ mod protocol;
 mod server;
 
 fn main() {
-    env_logger::init().unwrap();
+    let logger_config = fern::DispatchConfig {
+        format: Box::new(|msg: &str, level: &log::LogLevel, _location: &log::LogLocation| {
+            format!("[{}][{}] {}", time::now().strftime("%Y-%m-%d][%H:%M:%S").unwrap(), level, msg)
+        }),
+        output: vec![fern::OutputConfig::stdout(), fern::OutputConfig::file("sccache2.log")],
+        level: log::LogLevelFilter::Trace,
+    };
+    if let Err(e) = fern::init_global_logger(logger_config, log::LogLevelFilter::Trace)
```

Comment:
This is a good start, but I don't think we want to log at `Trace` by default, that's very verbose. I guess to do anything useful we'll have to provide some way to set the log levels, whether via environment variable (like `env_logger` does) or commandline arguments. Maybe you should set these to `Warn` as the default right now, and in a follow-up change we can figure out how to make it easy to set the log levels?